src/views/Sidebar/ChatListIcon.tsx

```diff
@@ -0,0 +1,55 @@
+import { ReactElement, useState } from "react";
+import { IconButton } from "@mui/material";
+import ThemedIcon from "@/components/ThemedIcon";
+import TrashIcon from "@/assets/icons/trashBin.svg?react";
+import KebabIcon from "@/assets/icons/kebab.svg?react";
+import { useDeleteChatMutation } from "@/store/api";
+import { useLocation } from "react-router-dom";
+import RouteNames from "@/routes/routes-names";
+import NewChatPopup from "@/views/Sidebar/NewChatPopup";
+
+type ChatListIconProps = {
+  chatId: number;
+  name: string;
+};
+function ChatListIcon({ chatId, name }: ChatListIconProps): ReactElement {
+  const [deleteChat] = useDeleteChatMutation();
+  const [isPopupVisible, setIsPopupVisible] = useState(false);
+  const location = useLocation();
+
+  let icon = <TrashIcon />;
```

Comment:
- [ ]  could you find some elegant way for the lines 20-38?

Reply:
Yeah, I didn't like it either. I'll think about how to make that logic prettier.